### DISCUSSION TITLE
On-boarding PR with organisational presets

### DISCUSSION THREAD

**Discussion by 56KBs** [renovatebot/renovate#9781] [Request Help]
I'm trying to on-board a new project, however Renovate does not raise an on-boarding PR due tot he fact it has no dependencies it can update.

I am using organisational presets which would have caused Renovate to notice a dependency exists and needs updating - Is it expected for the on-boarding logic to not use the organisational preset?

**Comment by rarkins** [comment]
How are the organizational presets provided - directly in the `config.js` root level or inside `onboardingConfig`?

We should also maybe add a setting for this which says "onboard repos even if no dependencies found", but defaulting to false.

One other suggestion if this is just a one-off: I think if you  manually create a branch `renovate/config` with `renovate.json` in it then it might be enough to trigger the onboarding PR.

**Reply by 56KBs** [reply]
In our case we have the organisational preset within a repository in the same GitLab group:

- `group1/repo1`
- `group1/renovate-config`
    - This contains a `renovate.json` which allows Renovate to identify additional dependencies

I have considered just creating the `renovate.json` manually for the repository so renovate starts working, however I was hoping to not have to do this for each new repository created that doesn't happen to have any dependencies identified by the default `config.js`

**Reply by rarkins** [reply]
Is `group1/renovate-config` included as a preset in your `config.js` anywhere?

**Reply by rarkins** [reply]
I *think* you could probably add something to your `config.js` to make this happen (e.g. is it a `fileMatch` string you are adding to detect dependencies, or something else?) but also opened #9783

**Reply by 56KBs** [reply]
> Is `group1/renovate-config` included as a preset in your `config.js` anywhere?

No it is not included anywhere, however when on-boarding does occur, the `renovate.json` will have this repository listed in the `extends` section

**Reply by rarkins** [reply]
oh ok, so it's autodiscovered..

When onboarding happens IIRC this is how we do it:
- Scan the base branch and see if any deps
- If any deps found then create a branch, commit the default config, scan again on this branch, and create the onboarding PR

If that's right, then it's the first step which is the problem because it's scanning without applying your onboarding config at first. It could maybe be a quick fix to inject the detected onboarding config into the first step's scan.

**Reply by 56KBs** [reply]
Having looked at the code - It appears you're right. The on-boarding config definitely isn't applied during the initial scan.

I'm happy to look at implementing the changes needed to ensure the auto-discovered on-boarding configuration is loaded for the first scan, if it is reasonable default behaviour for Renovate

**Reply by viceice** [reply]
Yes, we like to see the onboarding pr to use the present, so best it open an issue first, so we can assign it to you.

**Reply by 56KBs** [reply]
Raised #9851 to cover this - I will create an example repository tomorrow
